floris/wind_data.py

```diff
@@ -482,48 +508,61 @@ def upsample(self, wd_step=None, ws_step=None, method="linear", inplace=False):
-        # (this avoids interpolation errors)
-        if len(wind_direction_column) == 1:
-            wind_direction_column = np.array(
-                [
-                    wind_direction_column[0] - 1,
-                    wind_direction_column[0],
-                    wind_direction_column[0] + 1,
-                ]
-            )
-            ti_matrix = np.vstack((ti_matrix, ti_matrix[0, :], ti_matrix[0, :]))
-            freq_matrix = np.vstack((freq_matrix, freq_matrix[0, :], freq_matrix[0, :]))
-            if self.value_table is not None:
-                value_matrix = np.vstack((value_matrix, value_matrix[0, :], value_matrix[0, :]))
-
-        # If the wind_speed column has length 1, then pad the wind_speed column with
-        # that value + and - 1
-        # and expand the matrices accordingly (this avoids interpolation errors)
-        if len(wind_speed_column) == 1:
-            wind_speed_column = np.array(
-                [wind_speed_column[0] - 1, wind_speed_column[0], wind_speed_column[0] + 1]
-            )
-            ti_matrix = np.hstack((ti_matrix, ti_matrix[:, 0][:, None], ti_matrix[:, 0][:, None]))
-            freq_matrix = np.hstack(
-                (freq_matrix, freq_matrix[:, 0][:, None], freq_matrix[:, 0][:, None])
+        # Make sure everything sorted by wind direction and wind speed
+        # This should be the case but it's possible user can pass in unsorted data
+        sort_indices_wd = np.argsort(wind_direction_column)
+        wind_direction_column = wind_direction_column[sort_indices_wd]
+        sort_indices_ws = np.argsort(wind_speed_column)
+        wind_speed_column = wind_speed_column[sort_indices_ws]
+        ti_matrix = ti_matrix[sort_indices_wd, :]
+        ti_matrix = ti_matrix[:, sort_indices_ws]
+        freq_matrix = freq_matrix[sort_indices_wd, :]
+        freq_matrix = freq_matrix[:, sort_indices_ws]
+        if self.value_table is not None:
+            value_matrix = value_matrix[sort_indices_wd, :]
+            value_matrix = value_matrix[:, sort_indices_ws]
+
+        # Pad out the wind directions and wind speeds to the min and max ranges
+        # Add the min range to the wind direction column
+        wind_direction_column = np.append(wd_range_min_current, wind_direction_column)
+        ti_matrix = ti_matrix = np.vstack((ti_matrix[0, :], ti_matrix))
+        freq_matrix = np.vstack((freq_matrix[0, :], freq_matrix))
+        if self.value_table is not None:
+            value_matrix = np.vstack((value_matrix[0, :], value_matrix))
+
+        # Add the max range to the wind direction column
+        wind_direction_column = np.append(wind_direction_column, wd_range_max_current)
+        ti_matrix = np.vstack((ti_matrix, ti_matrix[-1, :]))
+        freq_matrix = np.vstack((freq_matrix, freq_matrix[-1, :]))
+        if self.value_table is not None:
+            value_matrix = np.vstack((value_matrix, value_matrix[-1, :]))
+
+        # Pad out the wind speeds
```

Comment:
This comment is redundant to the comment on line 525 (maybe just mention wind directions on line 525)